modules/ballerina-logging/src/main/java/org/ballerinalang/logging/formatters/BallerinaLogFormatter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.logging.formatters;
+
+import org.ballerinalang.logging.BLogManager;
+import org.ballerinalang.logging.util.BLogLevelMapper;
+
+import java.util.Date;
+import java.util.logging.Formatter;
+import java.util.logging.LogRecord;
+
+/**
+ * A custom log formatter for formatting the ballerina.log file
+ *
+ * @since 0.89
+ */
+public class BallerinaLogFormatter extends Formatter {
+
+    private static final String format = BLogManager.getLogManager().getProperty(
+            BallerinaLogFormatter.class.getCanonicalName() + ".format");
+
+    @Override
+    public String format(LogRecord record) {
+        String source;
+        if (record.getLoggerName().length() <= "ballerina.".length()) {
```

Comment:
In order to have a root logger for all ballerina programs/services, the names of the loggers used by the log API are named as "ballerina.<package-name-of-the-ballerina-program>". If we do not have such a root logger for user level logs of Ballerina, there's no easy way to assign handlers for the logs of Ballerina programs. We also can't print the source of the log as "ballerina.x.y.z". So to avoid that, the sub string "x.y.z" is taken.

The if condition is there to distinguish between programs defined in a package and the ones without a package.